### PR TITLE
[No QA] Create a readme for API details

### PR DESCRIPTION
**Pull request by tgolen**
### Details
This contains most of the answers I've been giving to questions from people.

### Fixed Issues
None

### Tests
None

### QA Steps
None

### CODE REVIEW

**Comment by tgolen** [comment]
bump @amyevans for review

**Review by amyevans — DISMISSED** [review]
Looks good, and reading this thoroughly helped cement a few things for me, so definitely appreciating the addition! I've left a few non-blocking comments, feel free to merge instead if you'd like!

File: README.md [hunk 1]
```diff
@@ -285,10 +287,11 @@ This application is built with the following principles.
     5. UI inputs push data to the server (React component -> Action -> XHR to server).
     6. Go to 1
     ![New Expensify Data Flow Chart](/web/data_flow.png)
-1. **Offline first**
+2. **Offline first**
+    - Be sure to read [OfflineFirst.md](./OfflineUX.md)!
     - All data that is brought into the app and is necessary to display the app when offline should be stored on disk in persistent storage (eg. localStorage on browser platforms). [AsyncStorage](https://reactnative.dev/docs/asyncstorage) is a cross-platform abstraction layer that is used to access persistent storage.
     - All data that is displayed, comes from persistent storage.
-1. **UI Binds to data on disk**
+4. **UI Binds to data on disk**
```
[the remark this answers is not in the document]
Comment (tgolen) [reply README.md:294]: OK, sure. PHPStorm likes to automatically renumber them, and I agree with using all 1s. I'll update that.

**Comment by tgolen** [comment]
Updated

**Review by amyevans — APPROVED** [review]

**Pull request merged**